packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/camera.dart

```diff
@@ -110,42 +110,59 @@ class CameraPosition {
       'CameraPosition(bearing: $bearing, target: $target, tilt: $tilt, zoom: $zoom)';
 }
 
+/// Indicates which type of camera update this instance represents.
+enum CameraUpdateType {
+  /// New position for camera
+  newCameraPosition,
+  /// New coordinates for camera
+  newLatLng,
+  /// New coordinates bounding box
+  newLatLngBounds,
+  /// New coordinate with zoom level
+  newLatLngZoom,
+  /// Move by a scroll delta
+  scrollBy,
+  /// Zoom by a relative change
+  zoomBy,
+  /// Zoom to an absolute level
+  zoomTo,
+  /// Zoom in
+  zoomIn,
+  /// Zoom out
+  zoomOut,
+}
+
 /// Defines a camera move, supporting absolute moves as well as moves relative
 /// the current position.
-class CameraUpdate {
-  const CameraUpdate._(this._json);
+abstract class CameraUpdate {
```

Comment:
All of the changes here need Dart unit tests. I would recommend one new test for each factory that validates:
- that the class is as expected,
- that the `updateType` is as expected, and
- that the properties were passed through correctly.